src/app.js

```diff
@@ -5,12 +5,28 @@ const cors = require("cors");
 const controllers = require("./controllers");
 const { errors } = require("celebrate");
 
+const whitelist = require("../lib/config/index.js").whitelist;
+
+var corsOptions = {
+  origin: function (origin, callback) {
+    if (whitelist.indexOf(origin) !== -1) {
+      callback(null, true);
+    } else {
+      callback(new Error('Not allowed by CORS'));
+    }
+  }
+}
+
 const app = express();
 app.use(helmet());
 app.use(cors());
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
-app.use("/api/v1/", controllers);
+
+// https://github.com/expressjs/cors/issues/71
+app.use(function(req,res,next){ req.headers.origin = req.headers.origin || req.headers.host; next(); })
```

Comment:
Do we need full function or only
```
app.use(cors(corsOptions));
```

Reply:
Please use `app.use(cors(corsOptions))` or else cors is not used.

Reply:
We should be using controllers as well in the same call?

`app.use("/api/v1/", cors(corsOptions), controllers);`

or do you mean something else?

Reply:
The line you have commented on, pastes `host` to `origin` from headers.

Reply:
I don't think we need `app.use(function(req,res,next){ req.he`...

I believe you are changing `origin` on the header to be the value of `host` if the `origin` is no present. `host` is completely different.

Remove that and make the additional whitelist change and this is good to go.